index.js

```diff
@@ -30,32 +30,47 @@ module.exports = {
   treeForPublic(tree) {
     let trees = [];
     const targetBrand = this.options['@embroider/macros'].setOwnConfig.brand;
-    const pkgName = this.parent.pkg.name;
-    const isEngine = checkIfEngine(this.parent.pkg);
+    const ebmCandidates = this.parent.addons
+      .map((addon) => addon.addons)
+      .flat()
+      .filter((a) => a.addons.find((addon) => addon.pkg.name.includes('@upfluence/ember-brand-manager')))
+      .reduce(
+        (acc, v) => {
+          console.log('<< fo', v.name, v.pkg.root);
+          acc.push({ pkgName: v.name, isEngine: true, srcDir: v.pkg.root });
+          return acc;
+        },
+        [{ pkgName: this.parent.pkg.name, isEngine: checkIfEngine(this.parent.pkg) }]
+      );
 
-    debugLog(`\n[EBM] Target brand : [${targetBrand}]`);
-    debugLog(`[EBM] Parent pkg name : [${pkgName}]`);
     if (tree) {
       trees.push(tree);
     }
 
-    this.setPublicAssets(trees, DEFAULT_BRAND, pkgName, isEngine);
+    debugLog(`\n[EBM] Target brand : [${targetBrand}]`);
 
-    if (targetBrand !== DEFAULT_BRAND) {
-      this.setPublicAssets(trees, targetBrand, pkgName, isEngine);
-    }
+    ebmCandidates.forEach((ebmCandidate) => {
+      debugLog(`[EBM] Pkg name : [${ebmCandidate.pkgName}]`);
+
+      this.setPublicAssets(trees, DEFAULT_BRAND, ebmCandidate.pkgName, ebmCandidate.isEngine, ebmCandidate.srcDir);
+
+      if (targetBrand !== DEFAULT_BRAND) {
+        this.setPublicAssets(trees, targetBrand, ebmCandidate.pkgName, ebmCandidate.isEngine, ebmCandidate.srcDir);
+      }
```

Comment:
```suggestion
      const brand = targetBrand === DEFAULT_BRAND ? DEFAULT_BRAND : targetBrand
      this.setPublicAssets(trees, brand, ebmCandidate.pkgName, ebmCandidate.isEngine, ebmCandidate.srcDir);
```
Maybe you can compact this one with that

Reply:
⚠️ 
We can't do that as we also need the `setPublicAssets` method to be run separately with the `DEFAULT_BRAND`

Reply:
My bad, I misunderstood the code, we can resolve this comment